packages/benchmark/run-tests.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/sh
+
+set -e 
+
+echo $(yarn --version)
+
+are_engines_installed() {
+    [ -d "$HOME/.esvu/engines/xs" ] && [ -d "$HOME/.esvu/engines/v8" ]
+}
+
+if are_engines_installed; then
+    echo "Engines already installed. Skipping installation."
+else
+    echo "Installing engines..."
+    INSTALL_OUTPU_XS=$(yarn dlx esvu install xs 2>&1) || INSTALL_STATUS_XS=$?
+    INSTALL_OUTPU_V8=$(yarn dlx esvu install v8 2>&1) || INSTALL_STATUS_V8=$?
```

Comment:
I'm not comfortable with `yarn test` modifying $HOME.
```suggestion
    echo "xs and/or v8 not found in $HOME/.esvu/engines; please esvu install them."
    # The bash "command not found" exit status code is 127.
    # https://www.gnu.org/software/bash/manual/html_node/Exit-Status.html#Exit-Status-1
    exit 127
```

Reply:
Changing ci.yaml is _entirely_ appropriate; this adds a new requirement on the host environment just like the one currently fulfilled by setup-node.

Reply:
And for local purposes, a package.json script for use like `yarn run install-engines` would make the consequences more clear.

Reply:
I am still having second thoughts about this change. Although, what you are saying is right, but we know that there are follow-up tasks aligned. This code might not even exist in the next PR.  So, we have to reverse the `ci.yaml` again? 

Another reason is `ci.yaml` is being run for every package in the `package/` directory. So, it will be unnecessary side effects to install `esvu` and `eshost` globally. 

@turadg had a similar idea in his mind to not install `esvu` and `eshost` globally. [Link to the comment](https://github.com/endojs/endo/pull/2720#issuecomment-2664210550).


I am more inclined towards having different script (as you mentioned in the second comment).
> And for local purposes, a package.json script for use like yarn run install-engines would make the consequences more clear.

* `yarn install-engines` which will do all the `esvu` and `eshost` related stuff.
* `yarn test:bench` to run the benchmark tests. In this case bundle the code and run the tests in `index.test.js`
* `yarn test` to run `yarn install-engines` and `yarn test:bench`. Might be vague in names but open for suggestions.

Reply:
But, even with different scripts `yarn` will be modifying the `$HOME` directory.

Reply:
> I am still having second thoughts about this change. Although, what you are saying is right, but we know that there are follow-up tasks aligned. This code might not even exist in the next PR. So, we have to reverse the `ci.yaml` again?

Yes. You can add a comment as a reminder.

> Another reason is `ci.yaml` is being run for every package in the `package/` directory. So, it will be unnecessary side effects to install `esvu` and `eshost` globally.

That's not what I'm seeing at https://github.com/endojs/endo/blob/master/.github/workflows/ci.yml. Only the [job "test"](https://github.com/endojs/endo/blob/master/.github/workflows/ci.yml#L58-L95) runs `yarn test` at package scope, and it does so after some setup steps. So what I'm asking for is insertion of esvu engine installation and eshost setup before its `yarn build` step, which will run exactly as often as the rest of those steps (i.e., once for each item of the (`[node-version](node-version: [18.x, 20.x, 22.4.1])`, `platform: [ubuntu-latest]`) `matrix` cross product).

> I am more inclined towards having different script (as you mentioned in the second comment).

Having the separate package.json script is good, and the CI job step could even use it to reduce duplication. But the step on its own does not resolve this issue, because CI _must_ have the engines but local `yarn test` _must not_ install them.

> > And for local purposes, a package.json script for use like yarn run install-engines would make the consequences more clear.
> 
> * `yarn install-engines` which will do all the `esvu` and `eshost` related stuff.
> * `yarn test:bench` to run the benchmark tests. In this case bundle the code and run the tests in `index.test.js`
> * `yarn test` to run `yarn install-engines` and `yarn test:bench`. Might be vague in names but open for suggestions.

No, I'm still objecting to `yarn test` modifying $HOME.